llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -588,6 +588,22 @@ static Instruction *foldCttzCtlz(IntrinsicInst &II, InstCombinerImpl &IC) {
     }
   }
 
+  // cttz(Pow2) -> Log2(Pow2)
+  // ctlz(Pow2) -> BitWidth - 1 - Log2(Pow2)
+  if (IsTZ || II.hasOneUse()) {
```

Comment:
Why does the number of uses of the *intrinsic* matter?

Reply:
Oops. I guess @goldsteinn means `Op0->hasOneUse()`.

Reply:
should probably be no multi-use check. We are replacing `ctlz` with `sub` which is just worth it.

Reply:
Your new condition effectively drops ctlz entirely :)

Reply:
:/ fixed....